crates/bevy_render/src/render_ecs_resource.rs

```diff
@@ -0,0 +1,44 @@
+use std::marker::PhantomData;
+
+use bevy_app::{App, Plugin};
+use bevy_ecs::system::{Commands, Res, Resource};
+pub use bevy_render_macros::ExtractResource;
+
+use crate::{RenderApp, RenderStage};
+
+/// Describes how a resource gets extracted for rendering.
+///
+/// Therefore the resource is transferred from the "main world" into the "render world"
+/// in the [`RenderStage::Extract`](crate::RenderStage::Extract) step.
+pub trait ExtractResource: Resource {
```

Comment:
I think we should try making this follow the `ExtractComponent` pattern (which is roughly the `From<T>` pattern). In theory, not all resources should be directly cloned into the same type (although in practice we do tend to do this).

```rust
pub trait ExtractResource: Resource {
    type Source: Resource;
    fn extract_resource(source: &Self::Source) -> Self;
}
```

I think this could still work with the `derive(ExtractResource)` pattern, which could assume Clone / `Self::Source = Self`

Reply:
Done.